Show archive icon for archived channels and add test (fixes #1992)

nsamuelreddy opened the pull request:
This PR updates the channel list to display an archive SVG icon for channels marked as archived.

**Changes:**
- Shows the archive icon for archived channels and the normal icon for others
- Registers the SVG asset in `pubspec.yaml` and places it in `assets/icons/archive.svg`
- Adds a widget test to verify the archive icon appears for archived channels and a normal icon for non-archived channels

**Testing:**
- Ran the new widget test to confirm correct icon display

Fixes #1992

chrisbobbe reviewed (commented):
Thanks!

Please add the icon in a separate commit, saying where it came from. For similar commits which you can use as examples, please browse the commit history (instructions [here](https://github.com/zulip/zulip-mobile/blob/main/docs/howto/git.md#2-the-secret-to-using-git-log--p)), and be sure to format both commit messages according to the project style: https://github.com/zulip/zulip-mobile/blob/master/docs/style.md#git-commits-commit-messages

nsamuelreddy commented on lib/widgets/all_channels.dart:
```diff
@@ -109,7 +109,7 @@ class AllChannelsListEntry extends StatelessWidget {
               size: 20,
               color: colorSwatchFor(context, subscription).iconOnPlainBackground,
-              iconDataForStream(channel)),
+              channel.isArchived ? ZulipIcons.archive : iconDataForStream(channel)),
```
Thanks! I’ll move the icon addition into a separate commit and include its source, and update the commit messages to follow the project style.

chrisbobbe commented:
Thank you! We now have a custom icon to use; please copy-paste the SVG from here and update the PR: [#mobile-design > archived-channel icon @ 💬](https://chat.zulip.org/#narrow/channel/530-mobile-design/topic/archived-channel.20icon/near/2331344)

nsamuelreddy commented:
Thanks, I've updated the archive icon with Karl's custom SVG from the mobile-design discussion and regenerated the icon font. The PR is now updated with the custom Zulip design.

chrisbobbe commented on lib/widgets/icons.dart:
```diff
@@ -24,179 +24,182 @@ abstract final class ZulipIcons {
   // BEGIN GENERATED ICON DATA
 
+  /// The Zulip custom icon "archive".
+  static const IconData archive = IconData(0xf101, fontFamily: "Zulip Icons");
```
How did you create the link to the design discussion, in the commit message? The channel-ID part is wrong, so it doesn't take me to the intended message.

chrisbobbe commented on lib/widgets/icons.dart:
```diff
@@ -206,6 +209,7 @@ IconData iconDataForStream(ZulipStream stream) {
   //   see this message and the one after it:
   //   https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/design.3A.20.23F117.20.22Inbox.22.20screen/near/1680637
   return switch(stream) {
+    ZulipStream(isArchived: true) => ZulipIcons.archive,
```
Please don't start showing the icon in the same commit that adds the asset to the project. I mentioned this in my last review: https://github.com/zulip/zulip-flutter/pull/2037#pullrequestreview-3580155281

chrisbobbe commented on test/widgets/all_channels_test.dart:
```diff
@@ -309,4 +309,19 @@ void main() {
+
+    final activeIcons = tester.widgetList<Icon>(find.byType(Icon))
+      .where((icon) => icon.icon != ZulipIcons.archive);
+    check(activeIcons).isNotEmpty();
```
Let's use `check(…).findsOne()` for both of these; see other tests for examples.

chrisbobbe commented:
> For screenshots - I'm having trouble getting archived channels to sync to the app. I archived a test channel on the web, but it's not appearing in the mobile app even after restarting. The feature works correctly in the automated tests though. Any suggestions on how to get an archived channel to show up for screenshots?

One option is to make temporary code changes, to pretend that a non-archived channel is actually archived.

nsamuelreddy commented:
Fixed the commit message to link to Karl's specific message containing the SVG code (near/2331344).

Screenshots:

**Before** (current behavior - normal channel icons):
[BEFORE screenshot]
![Screenshot_2025-12-25-18-34-20-46_e2eb6dad386ea79a8c34fb979200da81.jpg](https://github.com/user-attachments/assets/4835f75c-be42-4ce9-bfd0-2450133196fc)



**After** (with this PR - archive icon on channels):
[AFTER screenshot]
![Screenshot_2025-12-25-18-00-23-33_e2eb6dad386ea79a8c34fb979200da81.jpg](https://github.com/user-attachments/assets/1e4e4f52-9d43-49e3-a7d7-0ed5263295d3)



The archive icon now displays correctly for archived channels.

nsamuelreddy commented:
Sure, I’ll add before-and-after screenshots of the channel action sheet and format them in a GitHub Markdown table to save vertical space, like in #1999

chrisbobbe commented:
The icon looks very slightly too high in the channel action sheet, to me; can you nudge it down just a little bit?

nsamuelreddy commented:
Sure — I’ll nudge the archive icon down slightly in the channel action sheet and update the alignment.

nsamuelreddy commented:
Hi chrisbobbe,
Just following up on my PR  regarding the icon size change. I’ve been waiting for a few days and wanted to check if it can be merged or if any changes are needed.
Thanks!

chrisbobbe commented on test/widgets/all_channels_test.dart:
```diff
@@ -309,4 +309,14 @@ void main() {
+    await tester.pump();
+
+    check(find.byWidgetPredicate((widget) =>
+      widget is Icon && widget.icon == ZulipIcons.archive)).findsOne();
```
nit: Use `find.byIcon` instead

chrisbobbe commented:
Thanks! LGTM, but please rebase atop current `main` (GitHub is saying there are conflicts to be resolved). Marking for Greg to review after you've done that.

nsamuelreddy commented:
Yes, the archive icon does show in the real app for archived channels. The screenshots I shared earlier were just to demonstrate what the icon looks like - I temporarily modified the code to make all channels display as archived for those screenshots. In normal usage, only channels where isArchived: true will show the archive icon.

I've fixed the formatting (switch (stream) with space) and changed the commit message prefix to channel: as you suggested.

gnprice commented:
> Yes, the archive icon does show in the real app for archived channels.

OK. What are the circumstances where that happens, then?

So we don't lose track: one of my comments above is still open, too.

gnprice commented:
OK. Can you show a screen in the app that has such a channel?

There are plenty of archived channels on chat.zulip.org, so you can look there. If you need help finding those channels, look in the Zulip help center.

nsamuelreddy commented:
I couldn't find archived channels in the app because the Flutter client doesn't declare the archived_channels capability yet, so the server doesn't send them. The code comment says: "Servers that don't send this property will only send non-archived channels."

The icon rendering works (as shown in the test), but we need to add the client capability declaration to receive archived channels from the server. Should I add that in this PR or separately?
